### PR TITLE
feat(contract): includes support

### LINKED ISSUE #203: enable compilation / usage of contracts that have includes of other files
I am not sure if this is supported yet. currently I guess I have to either include everything in one file or create a separate file with contract interface only to be able to interact with the contract.

### PR DESCRIPTION
**Pull request by davidyuk**
closes #203 
found issues https://github.com/aeternity/aeproject/issues/433 https://github.com/aeternity/aeproject/issues/435

### CODE REVIEW

File: src/actions/contract.js [hunk 1]
```diff
@@ -21,40 +21,48 @@
...
   descrPath, contractAddress, contractSource, contractBytecode, contractAci,
 }, { dummySource, descrMayNotExist } = {}) {
   let descriptor = {};
   if (descrPath && (!descrMayNotExist || await fs.exists(resolve(descrPath)))) {
     descriptor = await fs.readJson(resolve(descrPath));
   }
+  if (contractSource) {
+    const contractSourcePath = resolve(contractSource);
+    descriptor.source = (await fs.readFile(contractSourcePath)).toString();
+    // TODO: remove after fixing https://github.com/aeternity/aeproject/issues/435
```
Comment (thepiwo) [inline src/actions/contract.js:41]: maybe you can elaborate the need, maybe I missunderstood the "debug output" nature

File: package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
   },
   "dependencies": {
     "@aeternity/aepp-sdk": "^12.1.2",
+    "@aeternity/aeproject": "^4.1.5",
```
Comment (dincho) [inline package.json:10]: isn't this cycle dep?

Comment (marc0olo) [reply package.json:10]: I also thought about that. but probably NPM doesn't care 😆

Comment (thepiwo) [reply package.json:10]: no, aeproject doesn't depend on cli

Comment (marc0olo) [reply package.json:10]: I think dincho and I are talking about the SDK dependency. at least I was thinking about that one

Comment (thepiwo) [reply package.json:10]: the sdk doesn't depend on cli either

Comment (marc0olo) [reply package.json:10]: had a wrong thinking on that. nevermind 😛

File: src/actions/contract.js (hunk 1 above)
Comment (davidyuk) [reply src/actions/contract.js:41]: In this case, with `--json` argument cli's output should be a valid json object. If I use `utils.getFilesystem` as is then I get json object prefixed with "Adding include to filesystem" message that can't be parsed as json.

File: package.json (hunk 2 above)
Comment (davidyuk) [reply package.json:10]: For sure we already have one: calldata -> cli -> sdk -> calldata

I don't see problems with cycle dependencies, I assume as far as all packages compatible with non-breaking updates (uses `^` in version number), then npm would be able to install a single version of each dependency.

Later I would copy `utils.getFilesystem` to SDK

Comment (dincho) [reply package.json:10]: calldata does not depends on the cli perse, it uses it in some helper scripts.

My comment was about aeproject, I was assuming it uses the CLI. Anyway dependancy on aeproject still sounds weird to me :)

**Review by thepiwo — APPROVED** [review]
good for me if it works for you, is it in some backlog to move that function into the sdk?

**Pull request merged**